<h1 style='text-align: center;'>How to Use Python image_split Package</h1>
<h3 style='text-align: center;'>Shouke Wei, Ph.D. Professor<sup>1,2</sup></h3>
<h5 style='text-align: center;'><sup>1</sup> Deepsim Intelligence Technology Inc, BC V2T0G9, Abbotsford, Canada</h5>
<h5 style='text-align: center;'><sup>2</sup> Deepsim Academy, BC V2T0G9, Abbotsford, Canada</h5>
<h5 style='text-align: center;'>Email: shouke.wei@gmail.com</h5>

## 1. Brief introduction of the package

FThe **'image_split'** package provides a convenient tool for splitting image data into separate training, validation, and testing sets. With this package, you can easily manage the distribution of your image dataset for machine learning tasks. The package offers a simple function, **'image_split'**, that takes input and output directories, along with desired split ratios and an optional seed for randomization. The package is designed to streamline the process of data preparation, making it easier to organize and manage image datasets for model training and evaluation.

Developed by Shouke Wei, Ph.D. from Deepsim Academy, Deepsim Intelligence Technology Inc. (c) 2023

## 2. Installation
You can easily install it using **'pip'** from your terminal.

```python
pip install image_split
```

## 3. Example
In this example, let's download a folower dataset from Kaggle by <a href='https://www.kaggle.com/code/rajmehra03/flower-recognition-cnn-keras/input' target='_blank'>this link</a>. After download, unzip the file and put the **'flowers'** folder into your current working directory. 

In [2]:
# import split function from the package
from image_split import train_val_test_split

# Define the path of input data
input_dir = './flowers'
output_dir = './flowers_split'

# specify the split ratios
train_ratio = 0.8
test_ratio = 0.1
val_ratio = 0.1
seed = 42

# call the function to split the data
train_val_test_split(input_dir, output_dir, train_ratio, test_ratio, seed, val_ratio)

Total number of images: 4317
Number of images in train set: 3453
Number of images in validation set: 431
Number of images in test set: 431


The above results show that there are totally 4317 flower images. There are 3453 images for training, 431 images for validation set and the rest 431 images for testing.